modules/akka-http/src/main/scala/pureconfig/module/akkahttp/package.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig.module
+
+import akka.http.scaladsl.model.{IllegalUriException, Uri}
+import akka.http.scaladsl.model.Uri.ParsingMode
+import pureconfig.{ConfigReader, ConfigWriter}
+import pureconfig.error.CannotConvert
+
+import scala.util.Try
+
+package object akkahttp {
+
+  implicit val uriReader: ConfigReader[Uri] =
+    ConfigReader.fromString(str =>
+      Try(Uri(str, ParsingMode.Strict)).fold(
+        {
+          case err: IllegalUriException => Left(CannotConvert(str, "Uri", err.info.summary))
+          case err => Left(CannotConvert(str, "Uri", err.getMessage.replace('^', " ".charAt(0)).trim))
```

Comment:
Why are we parsing the message here, in the fallback exception case? If we have no information on why an exception was thrown we should use `ExceptionThrown` instead of `CannotConvert`.

Reply:
used `ExceptionThrown` for fallback cases.